src/components/Recorder/recorder.tsx

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { RecordingStatus } from '../RecordingStatus/recordingStatus';
+import { ReactMic } from 'react-mic';
+import { RecordingTimer } from '../Timer/timer';
+import './recorder.css';
+
+type recorderState = {
+  isRecording: boolean;
+  buttonText: string;
+};
+
+interface EmptyProps {
+}
+
+export class Recorder extends React.Component<EmptyProps, recorderState> {
+  constructor(props: EmptyProps, state: recorderState) {
+    super(props, state);
+    this.state = {
+      isRecording: false,
+      buttonText: 'Start',
+    };
+  }
+
+  toggleRecording() {
+    if (this.state.isRecording) {
+      this.setState({ isRecording: false, buttonText: 'Start' });
+      return;
+    }
+    this.setState({ isRecording: true, buttonText: 'Stop' });
+  }
+
+  onData(recordedBlob: any) {
+    console.log('chunk of real-time data is: ', recordedBlob);
+  }
+
+  onStop(recordedBlob: any) {
+    console.log('recordedBlob is: ', recordedBlob);
+  }
+
+  render() {
+    return (
+      <div className="container">
+        <RecordingStatus isRecording={this.state.isRecording} />
```

Comment:
Like above. Pull out the elements of the state that you need, before the return value.